node/src/reactor/joiner.rs

```diff
@@ -518,8 +518,14 @@ impl reactor::Reactor for Reactor {
                                 block_header_with_old_version,
                             }) => {
                                 let old_version = block_header_with_old_version.protocol_version();
-                                info!(%current_version, %old_version, "restarting for downgrade");
-                                Some(JoinerEvent::Shutdown(ExitCode::DowngradeVersion))
+                                fatal!(
+                                    effect_builder,
+                                    "network is still running an older version, exiting; current_version={}; old_version={}",
+                                    current_version,
+                                    old_version
+                                )
```

Comment:
Actually, is this much different than the general `Err(error)` case below? Maybe the whole `match` arm can just be removed?
(Either way, would be good to get a second approval, too.)

Reply:
Well, good catch. The error will be formatted by the `fatal!` macro with slightly different message:
```
    #[error(
        "Current version is {current_version}, but current block header has older version: \
         {block_header_with_old_version:?}"
    )]
    CurrentBlockHeaderHasOldVersion {
        current_version: ProtocolVersion,
        block_header_with_old_version: Box<BlockHeader>,
    },
```

My preference would be to delete the match arm (`LinearChainSyncError::CurrentBlockHeaderHasOldVersion`) and rephrase the error message in the code block above.

Reply:
I'd also delete this `match` arm.
Maybe the `Err(error)` case below should just use the errors' `Display` impl, not `Debug`. Then it would print a readable message in every case?